javascript/ql/src/Security/CWE-300/InsecureDependencyResolution.qhelp

```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>Using an insecure protocol like HTTP or FTP to download your dependencies leaves your npm build vulnerable to a
+<a href="https://en.wikipedia.org/wiki/Man-in-the-middle_attack">Man in the Middle (MITM)</a>.
+This can allow attackers to inject malicious code into the artifacts that you are resolving and infect build artifacts
+that are being produced. This can be used by attackers to perform a
+<a href="https://en.wikipedia.org/wiki/Supply_chain_attack">Supply chain attack</a>
+against your project's users.
+</p>
```

Comment:
```suggestion
that are being produced. If such an infected artifact is later published, it may become a part of a
<a href="https://en.wikipedia.org/wiki/Supply_chain_attack">supply chain attack</a> that likely executes malicious code on unsuspecting victims.
</p>
```

Perhaps the full link should be in the references section.

Reply:
Also worth mentioning: the dependencies may execute malicious code on the developer machine itself!